TACC/Core-CMS-Custom
# Task/wp 284 submitter listing table

## Overview

The goal of this task is to create a new submitter_renewal_listing app to show a listing of all registrations for a submitter. The app:

- checks the role for submitter_admin
- grabs the submitter code
- grabs all distinct registration_ids filtered with the submitter code so only that submitter's organization's registrations are shown. 


## Related

- [WP-284](https://jira.tacc.utexas.edu/browse/WP-284)


## Changes
- In apcd_database.py:

1. Changed function name to get_submitter_info(user) so that multiple apps can use it and it be clear what the function does
2. Changed the get_registrations, get_registration_entities, and get_registration_contacts so that if a submitter_id is passed to the function, it will join the the submitter table with the registration and get all registration ids with a the matching submitter code.

- Created new submitter_renewals_listing app. Copied and pasted the admin_regis_table - the javascript was too fragile to put conditionals in the admin_regis_table view and template. 
**_- REASON:_** Since there is a tight deadline, Tracy requested that the admin_regis_table be completely copied and modified into its own app. Since we are planning to rework into React, she requested that the reuse of code be done when we switch to React rather than now.
- Changed wording  of header and description per Jodie at UTH
- Reused admin view modal and removed submitter view modal
- Made SubmittersTable a sub class of registrationsTable view
- Added drop down to link to form (This doesn't work yet until it's merged with Garrett's PR)



## Testing

1. Change APCD DB to prod credentials - test DB does not have year as a column on the registrations table yet. Joe is working on this
2. Go to http://localhost:8000/register/list-registration-requests/
3. Ensure the page looks like the screen shot. It doesn't look like TACCTEST submitter code has any associated entities or contacts. 
4. Double check that http://localhost:8000/administration/list-registration-requests/ is working as it previously was.

## UI

![Screenshot 2023-10-25 at 12 40 34 PM](https://github.com/TACC/Core-CMS-Custom/assets/96220951/8be85adb-866f-4f6d-9f5f-de4a37f9ffc8)
